bl/blocks/src/diagonal.cpp

```diff
@@ -0,0 +1,10 @@
+#include "diagonal.hpp"
+
+std::list<std::pair<int, int>> Diagonal::getShape() const {
+    std::list<std::pair<int, int>> l;
+    l.push_back(std::pair<int, int>(0, 0));
```

Comment:
If you'd put the list as the class member and initlize it in constructor and here just return it here it would be much faster

Reply:
fixed